x-pack/plugins/ml/public/components/field_type_icon/field_type_icon_view.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { EuiToolTip } from '@elastic/eui';
+
+// don't use something like plugins/ml/../common
+// because it won't work with the jest tests
+import { ML_JOB_FIELD_TYPES } from '../../../common/constants/field_types';
+
+export function FieldTypeIcon({ tooltipEnabled = false, type }) {
+  let ariaLabel = '';
+  let iconClass = '';
+  let iconChar = '';
+
+  switch (type) {
```

Comment:
We should migrate these Font Awesome icons to EUI icons I think. There are suitable replacements for most of these:

- fa-adjust - invert
- fa-clock-o - clock
- hash symbol - number
- fa-file-text-o - document (although would be nice to have one for a text style document)

I can't find any suitable replacements for
- fa-globe
- fa-laptop

You should raise an issue in the eui repo for the icons we need added.